# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/OriginOperations.cpp
# Commit: 2fd502bdc2d9
# Full Hash: 2fd502bdc2d921b888fc6597b6b3d3514226e8a2
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-12 20:37:56
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Update the cached value in PersistOp::DoDirectoryWork only after successful creation of the metadata file; r=dom-storage-reviewers,jari
#   
#   Currently, the cached value is updated before creation of the metadata file
#   which can lead to a mismatch between the metadata file and the cached value if
#   creation of the metadata file fails. It will be safer to update the cached
# ==============================================================================

diff -r 8b98c798b362 -r 2fd502bdc2d9 dom/quota/OriginOperations.cpp
--- a/dom/quota/OriginOperations.cpp	Sat Oct 12 10:00:14 2024 +0000
+++ b/dom/quota/OriginOperations.cpp	Sat Oct 12 10:00:14 2024 +0000
@@ -3106,8 +3106,7 @@
     // Origin directory has been successfully created.
     // Create OriginInfo too if temporary storage was already initialized.
     if (aQuotaManager.IsTemporaryStorageInitializedInternal()) {
-      timestamp = aQuotaManager.NoteOriginDirectoryCreated(
-          originMetadata, /* aPersisted */ true);
+      timestamp = aQuotaManager.NoteOriginDirectoryCreated(originMetadata);
     } else {
       timestamp = PR_Now();
     }
@@ -3115,6 +3114,10 @@
     QM_TRY(MOZ_TO_RESULT(QuotaManager::CreateDirectoryMetadata2(
         *directory, timestamp,
         /* aPersisted */ true, originMetadata)));
+
+    if (aQuotaManager.IsTemporaryStorageInitializedInternal()) {
+      aQuotaManager.PersistOrigin(originMetadata);
+    }
   } else {
     QM_TRY_INSPECT(
         const bool& persisted,